### PR TITLE
Modernize and cleanup cookbook

### PR DESCRIPTION
**Pull request by ramereth**
# Description

Describe what this change achieves

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+# htpasswd Cookbook
 
 [![Cookbook Version](https://img.shields.io/cookbook/v/htpasswd.svg)](https://supermarket.chef.io/cookbooks/htpasswd)
 [![Build Status](https://img.shields.io/circleci/project/github/sous-chefs/htpasswd/master.svg)](https://circleci.com/gh/sous-chefs/htpasswd)
 [![OpenCollective](https://opencollective.com/sous-chefs/backers/badge.svg)](#backers)
 [![OpenCollective](https://opencollective.com/sous-chefs/sponsors/badge.svg)](#sponsors)
 [![License](https://img.shields.io/badge/License-Apache%202.0-green.svg)](https://opensource.org/licenses/Apache-2.0)
 
-## Description
+Manage an htpasswd file.  If htpasswd exe isn't found, we install a python implementation.
```
Comment (tas50) [inline README.md:9]:
```suggestion
Manage an htpasswd file. If htpasswd exe isn't found, we install a python implementation.
```

File: README.md [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
-Work on linux with python for build-in implementation
+### Platforms
+
+- CentOS 7+
+- Debian 9+
+- Ubuntu 16.04+ LTS
+
+### Chef
+
+This cookbook requires Chef >= 14.
```
Comment (tas50) [inline README.md:25]:
```suggestion
This cookbook requires Chef Infra >= 14.
```

File: kitchen.dokken.yml [hunk 3]
```diff
@@ -2,14 +2,13 @@
...
   privileged: true
-  chef_version: current
+  chef_version: <%= ENV['CHEF_VERSION'] || 'current' %>
 
 transport:
   name: dokken
 
 provisioner:
   name: dokken
-  deprecations_as_errors: true
```
Comment (tas50) [inline kitchen.dokken.yml:12]: It's probably worth leaving this so CI jobs fail if something new gets added that cookstyle can't catch

File: kitchen.dokken.yml [hunk 4]
```diff
@@ -68,8 +65,3 @@ platforms:
       pid_one_command: /bin/systemd
       intermediate_instructions:
         - RUN zypper --non-interactive install aaa_base perl-Getopt-Long-Descriptive which net-tools
```
Comment (tas50) [inline kitchen.dokken.yml:67]: which and net-tools shouldn't be necessary here anymore https://github.com/test-kitchen/dokken-images/blob/master/fedora-latest/Dockerfile#L31

File: recipes/default.rb [hunk 5]
```diff
@@ -18,4 +18,8 @@
 # limitations under the License.
 #
 
+build_essential 'htpasswd'
```
Comment (tas50) [inline recipes/default.rb:21]: do we want these to happen at compile time?

File: resources/default.rb [hunk 6]
```diff
@@ -2,8 +2,8 @@
 default_action :add
 
 attribute :file, kind_of: String, name_attribute: true
-attribute :user, kind_of: String, required: true
-attribute :password, kind_of: String, required: true
+attribute :user, kind_of: String, required: [:add]
```
Comment (tas50) [inline resources/default.rb:5]: This is probably going to break this for chef 14 now

File: kitchen.dokken.yml (hunk 3 above)
Comment (ramereth) [reply kitchen.dokken.yml:12]: Doesn't this get inherited from `kitchen.yml`?

File: kitchen.dokken.yml (hunk 4 above)
Comment (ramereth) [reply kitchen.dokken.yml:67]: You added this in a previous commit 😉

File: resources/default.rb (hunk 6 above)
Comment (ramereth) [reply resources/default.rb:5]: It worked when I tried it.

File: kitchen.dokken.yml (hunk 4 above)
Comment (tas50) [reply kitchen.dokken.yml:67]: Well good job me

File: kitchen.dokken.yml (hunk 3 above)
Comment (tas50) [reply kitchen.dokken.yml:12]: Oh yeah if you have it in the main config your GTG

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
I'm going to make a PR in actionshub/chef-delivery so we can install any dependent gems for running rspec. That should fix the delivery check error.

**Review by Xorima — APPROVED** [review]
We should release this ad 1.0.0 else we will never release a 1.0.0

**Pull request merged**